dht/dht.go

```diff
@@ -75,6 +90,12 @@ type ServerConfig struct {
 	IPBlocklist iplist.Ranger
 	// Used to secure the server's ID. Defaults to the Conn's LocalAddr().
 	PublicIP net.IP
+	// Hook functions to be called when matching KRPC queries arrive.
+	// Hooks run prior to default KRPC handling, and may modify messages
+	// or indicate whether to skip default handling.
+	// Hooks can therefore replace, extend, or disable KRPC query handling
+	// in a Query-type specific way.
+	KRPCHooks map[string]KRPCHook
```

Comment:
Do you mean that users should add a single function that handles any given thing they want, using something like the KRPCHook interface, and then indicates wholesale whether to continue with processing?

I could certainly change it to a single-function interface, though I feel like the map interface is easier to reason about and compose, for complex projects. It elides the redundant task of determining and dispatching the query by name, and discourages any potentially weird handler crossovers. But, if people want weird crossovers they can add KRPCHook-compliant methods of some custom struct/subrouter as easily as sole functions.

Still, it's your party; if you'd rather a single interface function I can refactor. Do you like the KRPCHook interface overall? Will comment on the suggestion of edit-msg-in-place vs. return-new-msg on that line comment.